extensions/opentelemetry/deployment/src/main/java/io/quarkus/opentelemetry/deployment/exporter/otlp/OtlpExporterProcessor.java

```diff
@@ -75,6 +77,23 @@ public boolean getAsBoolean() {
         }
     }
 
+    static class OtlpUpstreamExporterEnabled implements BooleanSupplier {
+        OtlpExporterBuildConfig exportBuildConfig;
+        OTelBuildConfig otelBuildConfig;
+
+        public boolean getAsBoolean() {
+            return otelBuildConfig.enabled() &&
+                    exportBuildConfig.enabled() &&
+                    ((otelBuildConfig.logs().enabled().orElse(Boolean.TRUE) &&
+                            otelBuildConfig.logs().exporter().contains(OTLP_VALUE)) ||
+                            (otelBuildConfig.metrics().enabled().orElse(Boolean.TRUE) &&
+                                    otelBuildConfig.metrics().exporter().contains(OTLP_VALUE))
+                            ||
+                            (otelBuildConfig.traces().enabled().orElse(Boolean.TRUE) &&
+                                    otelBuildConfig.traces().exporter().contains(OTLP_VALUE)));
```

Comment:
This is super hard to read. I propose breaking it up into the appropriate pieces and giving them proper variable names